src/HPolygon.jl

```diff
@@ -61,7 +61,7 @@ norm zero, any vertex is returned.
 Comparison of directions is performed using polar angles; see the overload of
 `<=` for two-dimensional vectors.
 """
-function σ(d::AbstractVector{<:Real}, P::HPolygon{N})::Vector{N} where {N<:Real}
+function σ(d::AbstractVector{N}, P::HPolygon)::Vector{N} where {N<:Real}
```

Comment:
alright; maybe add a line of this behavior in the output, or with an example .. a priori we could have matched the output numeric type with that of the input set (as it was before)..

Reply:
.. hmm why don't we keep the signature 

```julia
σ(d::AbstractVector{N}, P::HPolygon{N})::Vector{N} where {N<:Real}
```

Reply:
The problem is that for the interface tests I need consistent *input* types.
In this case I went for the minimal change, which was to remove the numeric type for the second argument for the polygon types.
The current signature is `σ(::AbstractVector{<:Real}, ::LazySet)`.

I am not against adding the type for the second argument again, but then we also need to add it for all other set types. Actually, I would prefer that.

What do you think?

Reply:
> I am not against adding the type for the second argument again, but then we also need to add it for all other set types. 

the motivation for `HPolygon` here could be to avoid the "unsafe" case where you pass an inexact vector `d` to an exact polygon `P`. 

i suppose that this type of observation holds for all other set types. but i also remember that we had some performance issues so i would be careful of changing everything; at least do it gradually.

Reply:
for instance in

```julia
function σ(d::V, lm::LinearMap) where {N<:Real, V<:AbstractVector{N}}
    return lm.M * σ(_At_mul_B(lm.M, d), lm.X)
end
```

we don't strictly fix that the numeric type of the `lm` is `N`, on purpose (the discussion is somewhere.)

Reply:
I think the problem was with the *output* type `AbstractVector{N}`, which we have dropped. I do not see a possible problem with the input type.

Reply:
Then, should I do the changes in this PR or in a new one? If in a new one, merge this one first or wait for the other one?

Reply:
> I think the problem was with the output type AbstractVector{N}, which we have dropped.

OK. Typically we have `σ(d::V, B::Ball1{N}) where {N<:Real, V<:AbstractVector{N}}` which is "safe" in the context of the current discussion. I guess that the types that need to be adapted are not that many after all.

> Then, should I do the changes in this PR or in a new one? If in a new one, merge this one first or wait for the other one?

I don't think that #468 is representative of the proposed change, so, if you want we can merge this branch and continue with #478.